Check response data type before appending

mycarrysun opened the pull request:
My implementation of responses sometimes has integers as the response content. This will of course produce an error trying to access an index of an integer like an array.

I have added a check to make sure the response data is an array before trying to attach the debug property to the response.

Also: I have added a small convenience for seeing the total time of all the queries in the query collection.

Commit:
Check if response data is array/object before adding debug data

Commit:
Add total time to query collection

mycarrysun committed:
Merge pull request #1 from mycarrysun/response-array-fix

Check response data type and total time on query collections

Commit:
Dont check if response data is an object

mycarrysun committed:
Merge pull request #2 from mycarrysun/response-array-fix

Dont check if response data is an object

Commit:
Add test for integer response

mycarrysun committed:
Merge pull request #3 from mycarrysun/response-array-fix

Add test for integer response

mlanin commented on tests/DebuggerTest.php:
```diff
@@ -199,4 +199,23 @@ public function it_can_show_cache_events()
                 ],
             ]);
     }
+
```
Please, add `/** @test */` annotation.
Without it, phpunit will skip it

Commit:
add @test annotation

mycarrysun committed:
Merge pull request #4 from mycarrysun/response-array-fix

add @test annotation

mycarrysun commented:
@mlanin added :)

mlanin commented:
Tests are failing now :(

mycarrysun commented:
@mlanin looks like for some reason the `$this->json()` method is returning the standard `Illuminate\Http\Response` type instead of the `TestResponse`, which makes sense why it can't find the `assertJsonMissingExact()` method.

Did I do anything wrong in the return type?

```php
/** @test */
public function is_does_not_add_debug_if_response_is_integer()
{
    $this->app['router']->get('foo', function () {
        lad('test if integer');

        return response()->json(1234);
    });

    $this->json('get', '/foo')
         ->assertStatus(200)
         ->assertJsonMissingExact([
            'debug' => [
                'dump' => [
                    'test if integer',
                ]
            ]
         ]);
}
```

mlanin commented:
Tests are failing for php 5.6 version. The highest laravel version that supports this is 5.4.36. But this version doesn't have `assertJsonMissingExact` method :(

mycarrysun commented:
Can we include this version requirement of laravel in composer.json then? The copy I have installed v5.7.28.

mlanin commented:
But this package supports 5.4 version. Your PR is really good, please, adjust your test and I'll merge it asap :)

Commit:
[feat] add disable helper to turn off debugger

Commit:
Merge branch 'enable-disable'

Commit:
Merge remote-tracking branch 'upstream/master'

# Conflicts:
#	tests/DebuggerTest.php

Commit:
update dependencies with php extensions and more strict versions

Commit:
check if data is an arr or object and return if not

Commit:
use assertJsonMissing instead of assertJsonMissingExact

mycarrysun commented:
> But this package supports 5.4 version. Your PR is really good, please, adjust your test and I'll merge it asap :)

I've updated the `composer.json` to use version of packages that are compatible with your requirements so composer doesn't try to install a newer version. I also added requirements for `sqlite` since it was missing from my machine and the tests were failing (Ubuntu 19).

Commit:
[fix] revert composer.json version constraints

Commit:
[fix] dont check for arrays, we only want objects

mlanin commented on composer.json:
```diff
@@ -11,12 +11,14 @@
     ],
     "require": {
+        "ext-sqlite3": "*",
+        "ext-pdo_sqlite": "*",
```
ext-sqlite3 is not a dependency for this package

mycarrysun replied:
Is it just pdo_sqlite then?

mlanin replied:
`pdo_sqlite` as well. This package doesn't use sqlite

mycarrysun replied:
Yes it does - that's why I added it. The unit tests use a sqlite database connection.

mycarrysun replied:
Maybe it should go in require-dev?

Commit:
move sqlite extensions to require-dev

Commit:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	composer.json

Commit:
use >= for version constraints in laravel dependencies

mycarrysun commented:
Closing from lack of activity from the maintainer. Feel free to reopen if you want it merged.
